### PR TITLE
Add checks for some XXX_New and XXX_Add functions

### PR DESCRIPTION
**Pull request by bmiklautz**

### CODE REVIEW

**Comment by hardening** [comment]
of course +1 :)

File: channels/encomsp/client/encomsp_main.c [hunk 1]
```diff
@@ -980,7 +1003,5 @@ BOOL VCAPITYPE VirtualChannelEntry(PCHANNEL_ENTRY_POINTS pEntryPoints)
 	encomsp->channelEntryPoints.pInterface = *(encomsp->channelEntryPoints.ppInterface);
 	encomsp->channelEntryPoints.ppInterface = &(encomsp->channelEntryPoints.pInterface);
 
-	encomsp_add_init_handle_data(encomsp->InitHandle, (void*) encomsp);
-
-	return 1;
+	return encomsp_add_init_handle_data(encomsp->InitHandle, (void*) encomsp) ? 1 : -1;
```
Comment (nfedera) [inline channels/encomsp/client/encomsp_main.c]: This function (VirtualChannelEntry) is BOOL. Just return encomsp_add_init_handle_data() - right ?

File: channels/rail/client/rail_main.c [hunk 2]
```diff
@@ -623,7 +641,5 @@ BOOL VCAPITYPE VirtualChannelEntry(PCHANNEL_ENTRY_POINTS pEntryPoints)
 	rail->channelEntryPoints.pInterface = *(rail->channelEntryPoints.ppInterface);
 	rail->channelEntryPoints.ppInterface = &(rail->channelEntryPoints.pInterface);
 
-	rail_add_init_handle_data(rail->InitHandle, (void*) rail);
-
-	return 1;
+	return rail_add_init_handle_data(rail->InitHandle, (void*) rail) ? 1 : -1;
 }
```
Comment (nfedera) [inline channels/rail/client/rail_main.c]: VirtualChannelEntry is BOOL

File: channels/encomsp/client/encomsp_main.c (hunk 1 above)
Comment (hardening) [inline channels/encomsp/client/encomsp_main.c]: Oops totally right

File: channels/rail/client/rail_main.c (hunk 2 above)
Comment (hardening) [inline channels/rail/client/rail_main.c]: Same error here !

**Comment by nfedera** [comment]
Superseeded by PR #2630

**Pull request closed without merging**